cli/flox/src/commands/build.rs

```diff
@@ -22,37 +23,84 @@ pub struct Build {
     #[bpaf(short('L'), long)]
     build_logs: bool,
 
-    /// The package to build, corresponds to the entries in
-    /// the 'build' table in the environment's manifest.toml.
-    /// If not specified, all packages are built.
-    #[bpaf(positional("build"))]
-    packages: Vec<String>,
+    #[bpaf(external(subcommand_or_build_targets))]
+    subcommand_or_targets: SubcommandOrBuildTargets,
+}
+
+#[derive(Debug, Bpaf, Clone)]
+enum SubcommandOrBuildTargets {
+    /// Clean the build directory
+    ///
+    /// Remove builds artifacts and temporary files.
+    #[bpaf(command, footer("Run 'man flox-build-clean' for more details."))]
+    Clean {
+        /// The package(s) to clean.
+        /// Corresponds to entries in the 'build' table in the environment's manifest.toml.
+        /// If not specified, all packages are cleaned up.
+        #[bpaf(positional("package"))]
+        targets: Vec<String>,
+    },
+    BuildTargets {
+        /// The package to build.
+        /// Corresponds to entries in the 'build' table in the environment's manifest.toml.
+        /// If not specified, all packages are built.
+        #[bpaf(positional("package"))]
+        targets: Vec<String>,
+    },
 }
 
 impl Build {
-    #[instrument(name = "build", skip_all)]
     pub async fn handle(self, config: Config, flox: Flox) -> Result<()> {
-        subcommand_metric!("build");
-
         if !config.features.unwrap_or_default().build {
             message::plain("🚧 👷 heja, a new command is in construction here, stay tuned!");
-            bail!("'build' feature is not enabled.");
+            bail!("'clean' feature is not enabled.");
         }
 
         let env = self
             .environment
-            .detect_concrete_environment(&flox, "Build")?;
+            .detect_concrete_environment(&flox, "Clean")?;
 
         if let ConcreteEnvironment::Remote(_) = &env {
-            bail!("Cannot build from a remote environment");
+            bail!("Cannot clean a remote environment");
         };
```

Comment:
(blocking, no need for re-review) Are these renames from an overeager search/replace or earlier refactor? "build" should still be the feature and verb, especially as we haven't parsed the intent at this point.